### PR TITLE
Unit Test For Defunct Processes Metric

### PR DESCRIPTION
**Pull request by raisaat**
#### What this PR does / why we need it:

This PR adds unit tests for the `crio_processes_defunct` metric. It also changes the function `DefunctProcesses` to `DefunctProcessesForPath` in `internal/process/defunct_processes.go` to take the path name of a filesystem as a parameter and adds `DefunctProcesses` as a wrapper for the function that passes the path name of the actual process filesystem, `/proc`, to it. This is needed for the unit tests as it allows to pass any filesystem for testing. 

#### Which issue(s) this PR fixes:
None

#### Special notes for your reviewer:
This PR adds a new directory `internal/process/testing` which consists of several other directories and a file to account for different test cases.
#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: internal/process/defunct_processes_test.go [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+import "testing"
+
+func TestDefunctProcessesForPath(t *testing.T) {
+	defunctCount, err := DefunctProcessesForPath("./testing/proc")
+	if err != nil {
+		t.Errorf("%v", err)
+	} else if defunctCount != 7 {
+		t.Errorf("DefunctProcessesForPath returned %d; should return 7.", defunctCount)
+	}
+}
```
Comment (saschagrunert) [inline internal/process/defunct_processes_test.go:12]: While this is fine, let's use the ginkgo test framework like we do in other tests within this repository. This means, we add a `suite_test.go` like this: https://github.com/cri-o/cri-o/blob/master/internal/config/apparmor/suite_test.go

Then we modify the `defunct_processes_test.go` to this basic skeleton:
```go
package process_test

import (
	. "github.com/onsi/ginkgo"
	. "github.com/onsi/gomega"

	"github.com/cri-o/cri-o/internal/process"
)

// The actual test suite
var _ = t.Describe("Process", func() {
	t.Describe("DefunctProcessesForPath", func() {
		It("should succeed", func() {
			defunctCount, err := process.DefunctProcessesForPath("./testing/proc")

			Expect(err).To(BeNil())
			Expect(defunctCount).To(Equal(7))
		})
	})
})
```
Let's also add a tests for the `err != nil` cases.

File: internal/process/defunct_processes_suite_test.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+	. "github.com/cri-o/cri-o/test/framework"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+// TestDefunctProcessesForPath runs the created specs
+func TestDefunctProcessesForPath(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunFrameworkSpecs(t, "DefunctProcesses")
```
Comment (saschagrunert) [inline internal/process/defunct_processes_suite_test.go:14]:
```suggestion
	RunFrameworkSpecs(t, "Process")
```

File: internal/process/defunct_processes_suite_test.go [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+import (
+	"testing"
+
+	. "github.com/cri-o/cri-o/test/framework"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+// TestDefunctProcessesForPath runs the created specs
+func TestDefunctProcessesForPath(t *testing.T) {
```
Comment (saschagrunert) [inline internal/process/defunct_processes_suite_test.go:12]: Let's assume we're adding multiple tests to this:

```suggestion
// TestProcess runs the created specs
func TestProcess(t *testing.T) {
```

**Review by saschagrunert — APPROVED** [review]
Just two nits, otherwise LGTM :+1:

**Review by TomSweeneyRedHat — COMMENTED** [review]
LGTM

**Review by saschagrunert — APPROVED** [review]
/lgtm
